File: rea2/app/src/main/java/com/jang/assistales/model/DeitiesBean.java
package com.jang.assistales.model;

import androidx.room.Entity;
import androidx.room.ForeignKey;
import androidx.room.PrimaryKey;

import java.io.Serializable;

@Entity(foreignKeys = @ForeignKey(entity = UniversBean.class, parentColumns = "id_univers", childColumns = "universId"))
public class DeitiesBean implements Serializable {
    @PrimaryKey
    private long id_deities;
    private String name;
    private String otherName;
    private String description;
    private String age;
    private String talent;
    private String weakness;
    private String iq;
    private String power;
    private String agility;
    private boolean group;
    private String tradition;
    private String skin;
    private String hair;
    private String eyes;
    private String ear;
    private String noze;
    private String height;
    private String weight;
    private String particularity;
    private long universId;

    @Override
    public String toString() {
        return name;
    }

//--------------------------Constructor---------------------------//

    public DeitiesBean(String name) {
        this.name = name;
    }


    //--------------------------Getter/Setter-------------------------//


    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    public String getOtherName() {
        return otherName;
    }

    public void setOtherName(String otherName) {
        this.otherName = otherName;
    }

    public String getDescription() {
        return description;
    }

    public void setDescription(String description) {
        this.description = description;
    }

    public String getAge() {
        return age;
    }

    public void setAge(String age) {
        this.age = age;
    }

    public String getTalent() {
        return talent;
    }

    public void setTalent(String talent) {
        this.talent = talent;
    }

    public String getWeakness() {
        return weakness;
    }

    public void setWeakness(String weakness) {
        this.weakness = weakness;
    }

    public String getIq() {
        return iq;
    }

    public void setIq(String iq) {
        this.iq = iq;
    }

    public String getPower() {
        return power;
    }

    public void setPower(String power) {
        this.power = power;
    }

    public String getAgility() {
        return agility;
    }

    public void setAgility(String agility) {
        this.agility = agility;
    }

    public boolean isGroup() {
        return group;
    }

    public void setGroup(boolean group) {
        this.group = group;
    }

    public String getTradition() {
        return tradition;
    }

    public void setTradition(String tradition) {
        this.tradition = tradition;
    }

    public String getSkin() {
        return skin;
    }

    public void setSkin(String skin) {
        this.skin = skin;
    }

    public String getHair() {
        return hair;
    }

    public void setHair(String hair) {
        this.hair = hair;
    }

    public String getEyes() {
        return eyes;
    }

    public void setEyes(String eyes) {
        this.eyes = eyes;
    }

    public String getEar() {
        return ear;
    }

    public void setEar(String ear) {
        this.ear = ear;
    }

    public String getNoze() {
        return noze;
    }

    public void setNoze(String noze) {
        this.noze = noze;
    }

    public String getHeight() {
        return height;
    }

    public void setHeight(String height) {
        this.height = height;
    }

    public String getWeight() {
        return weight;
    }

    public void setWeight(String weight) {
        this.weight = weight;
    }

    public String getParticularity() {
        return particularity;
    }

    public void setParticularity(String particularity) {
        this.particularity = particularity;
    }
}

File: rea5/assistales/app/src/main/java/com/jang/assistales/model/DeitiesBean.kt
package com.jang.assistales.model

import androidx.room.*

@Entity(
    tableName = "deities",
    /*foreignKeys = [ForeignKey(
        entity = UniversBean::class,
        parentColumns = ["id_univers"],
        childColumns = ["universId"]
    )]*/
)
class DeitiesBean(
    val name: String,
    @PrimaryKey(autoGenerate = true)
    val id_deities: Long = 0,
    val otherName: String? = null,
    val description: String? = null,
    val age: String? = null,
    val talent: String? = null,
    val weakness: String? = null,
    val iq: String? = null,
    val power: String? = null,
    val agility: String? = null,
    val Group: Boolean = false,
    val tradition: String? = null,
    val skin: String? = null,
    val hair: String? = null,
    val eyes: String? = null,
    val ear: String? = null,
    val noze: String? = null,
    val height: String? = null,
    val weight: String? = null,
    val particularity: String? = null
    /*val universId: Long = 0*/

) {
    override fun toString(): String {
        return name
    }

    @Dao
    interface DeitiesDao {
        @Query("SELECT * FROM deities ORDER BY name ASC")
        fun getAll(): List<DeitiesBean>

        @Insert
        fun insert(deities: DeitiesBean): Long

        @Update
        fun update(deities: DeitiesBean)

        @Delete
        fun delete(deities: DeitiesBean)
    }
}